cf-execd/cf-execd.c

```diff
@@ -434,6 +434,17 @@ void ThisAgentInit(void)
     umask(077);
 }
 
+// Return 2 to 62 seconds depending on how much time is left until the next
```

Comment:
Why 2 to 62 and not the actual remaining number of seconds?

Reply:
I wondered why 2 to 62 as well.

Reply:
Added more comments.

Reply:
It would be best to use the monotonic clock on platforms that have it.

Reply:
That's true, it would give better guarantees about how long we actually sleep. But still it would be a bit awkward, since the time you wake up would maybe not match what is expected. (Same as now).

Reply:
Yes, there are two separate things -- **how often** this code wakes up (monotonic clock) and **when** it happens (system time). We can only control the former, but the latter actually determines the defined classes and thus the agent run triggering. So... :man_shrugging: